functions/AuthUtils.js

```diff
@@ -0,0 +1,186 @@
+        path: '/',
+        maxAge: new Date(0),
+        httpOnly: true,
+    });
+};
+
+const generateNetlifyCookieFromAuth0Token = async (tokenData) => {
+    const netlifyToken = await generateNetlifyJWT(tokenData);
+
+    const twoWeeks = 14 * 24 * 3600000;
+    return cookie.serialize('nf_jwt', netlifyToken, {
+        secure: !process.env.NETLIFY_DEV === 'true',
+        path: '/',
+        maxAge: twoWeeks,
+    });
+};
+
+const getCallbackParams = (openIDClient, event) => {
+    /* NOTE: method, body, and url are all required for the openIDClient to work with
+    the request*/
+    const req = {
+        method: 'POST',
+        body: event.body,
+        url: event.headers.host,
+    };
+    //callbackParams documentation - https://github.com/panva/node-openid-client/tree/master/docs#clientcallbackparamsinput
+    return openIDClient.callbackParams(req);
+};
+
+const generateAuth0LogoutUrl = () => {
+    const auth0DomainLogout = `https://${process.env.AUTH0_DOMAIN}/v2/logout`;
+    const urlReturnTo = `returnTo=${encodeURIComponent(process.env.URL)}`;
+    const urlClientId = `client_id=${process.env.AUTH0_CLIENT_ID}`;
+    return `${auth0DomainLogout}?${urlReturnTo}&${urlClientId}`;
+};
+
+const handleLogin = async (event) => {
+    if (!event || !event.headers) {
+        throw new Error('Malformed event');
+    }
+    const openIDClient = await getOpenIDClient();
+    const referer = event.headers.referer;
+    const encodedStateStr = generateEncodedStateString(referer);
+    const nonce = generators.nonce();
+
+    //authorizationUrl docs - https://github.com/panva/node-openid-client/tree/master/docs#clientauthorizationurlparameters
+    const authRedirectURL = openIDClient.authorizationUrl({
+        scope: 'openid email profile',
+        response_mode: 'form_post',
+        nonce,
+        state: encodedStateStr,
+    });
+    return {
+        statusCode: 302,
+        headers: {
+            Location: authRedirectURL,
+            'Cache-Control': 'no-cache',
+            'Set-Cookie': generateAuth0LoginCookie(nonce, encodedStateStr),
+        },
+    };
+};
+
+const handleCallback = async (event) => {
+    if (!event || !event.headers || !event.headers.cookie) {
+        throw new Error('Invalid request');
+    }
+    const openIDClient = await getOpenIDClient();
+
+    const { auth0_login_cookie: loginCookie } = cookie.parse(
+        event.headers.cookie
+    );
+    const { nonce, state } = JSON.parse(loginCookie);
+
+    const params = getCallbackParams(openIDClient, event);
+
+    //callback docs - https://github.com/panva/node-openid-client/tree/master/docs#clientcallbackredirecturi-parameters-checks-extras
+    const tokenSet = await openIDClient.callback(
+        `${process.env.URL}/.netlify/functions/callback`,
+        params,
+        {
+            nonce,
+            state,
+        }
+    );
+    const { id_token } = tokenSet;
+    const decodedToken = jwt.decode(id_token);
```

Comment:
I think you can use `tokenset.claims()` here:

https://github.com/panva/node-openid-client/tree/master/docs#tokensetclaims